test/python/synthesis/test_clifford_sythesis.py

```diff
@@ -0,0 +1,282 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+"""Tests for Clifford synthesis functions."""
+
+import numpy as np
+from ddt import ddt
+from qiskit.circuit import Gate, QuantumCircuit, QuantumRegister
+from qiskit.circuit.library import (
+    CXGate,
+    CYGate,
+    CZGate,
+    DCXGate,
+    ECRGate,
+    HGate,
+    IGate,
+    SdgGate,
+    SGate,
+    SXGate,
+    SXdgGate,
+    SwapGate,
+    XGate,
+    YGate,
+    ZGate,
+    iSwapGate,
+)
+from qiskit.converters.dag_to_circuit import dag_to_circuit
+from qiskit.quantum_info.operators import Clifford
+from qiskit.synthesis.clifford import (
+    synth_clifford_full,
+    synth_clifford_ag,
+    synth_clifford_bm,
+    synth_clifford_greedy,
+)
+
+from test import QiskitTestCase  # pylint: disable=wrong-import-order
+from test import combine  # pylint: disable=wrong-import-order
+
+
+class VGate(Gate):
+    """V Gate used in Clifford synthesis."""
+
+    def __init__(self):
+        """Create new V Gate."""
+        super().__init__("v", 1, [])
+
+    def _define(self):
+        """V Gate definition."""
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q)
+        qc.sdg(0)
+        qc.h(0)
+        self.definition = qc
+
+
+class WGate(Gate):
+    """W Gate used in Clifford synthesis."""
+
+    def __init__(self):
+        """Create new W Gate."""
+        super().__init__("w", 1, [])
+
+    def _define(self):
+        """W Gate definition."""
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q)
+        qc.append(VGate(), [q[0]], [])
+        qc.append(VGate(), [q[0]], [])
+        self.definition = qc
+
+
+def random_clifford_circuit(num_qubits, num_gates, gates="all", seed=None):
+    """Generate a pseudo random Clifford circuit."""
+
+    qubits_1_gates = ["i", "x", "y", "z", "h", "s", "sdg", "sx", "sxdg", "v", "w"]
+    qubits_2_gates = ["cx", "cz", "cy", "swap", "iswap", "ecr", "dcx"]
+    if gates == "all":
+        if num_qubits == 1:
+            gates = qubits_1_gates
+        else:
+            gates = qubits_1_gates + qubits_2_gates
```

Comment:
This function still exists in the `test_cliffords` file, right? Maybe it would be nice to move it (and also `random_linear_circuit`) into the main `qiskit` repository. I don't know why we have not done this already, but more than once I wished we had this in the main codebase.